pkg/internal/checkup/trex/client.go

```diff
@@ -32,3 +46,111 @@ func NewClient(vmiSerialClient vmiSerialConsoleClient, namespace string, verbose
 		verbosePrintsEnabled: verbosePrintsEnabled,
 	}
 }
+
+func (t Client) WaitTrexServerReady(ctx context.Context, vmiName string) error {
```

Comment:
This method is called like this:
```go
trex.WaitTrexServerReady()
```

The second Trex is redundant IMO.

```suggestion
func (t Client) WaitForServerToBeReady(ctx context.Context, vmiName string) error {
```

Also, the vmiName could be part of the `Client` struct:
1. We currently support only a single traffic generator.
2. The parameter has the same value on every method call.

This comment is relevant for the other methods as well.

Reply:
Will change function name. DONE
About the vmiName param - totally - but let's do it in another PR.